handler.go

```diff
@@ -375,6 +376,59 @@ func (fm *Frontman) sendResultsChanToHubWithInterval(resultsChan chan Result) er
 	}
 }
 
+// HealthCheck runs before any other check to ensure that the host itself and its network are healthly.
+// This is useful to confirm a stable internet connection to avoid false alerts due to network outages.
+func (fm *Frontman) HealthCheck() error {
+	hcfg := fm.Config.HealthChecks
+	if len(hcfg.ReferencePingHosts) == 0 {
+		fm.HealthCheckNotPassed = false
+		return nil
+	}
+	count := hcfg.ReferencePingCount
+	if count == 0 {
+		fm.HealthCheckNotPassed = false
+		return nil
+	}
+	timeout := secToDuration(hcfg.ReferencePingTimeout)
+	if timeout == 0 {
+		// use the default timeout
+		timeout = 500 * time.Millisecond
+	}
+	successC := make(chan bool, len(hcfg.ReferencePingHosts))
+
+	wg := new(sync.WaitGroup)
+	for _, addr := range hcfg.ReferencePingHosts {
+		p, err := ping.NewPinger(addr)
+		if err != nil {
+			log.WithError(err).Warningln("failed to parse host for ICMP ping")
+			continue
+		}
+		p.Timeout = timeout
+		p.Count = count
+		p.OnRecv = func(_ *ping.Packet) {
+			successC <- true
+		}
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			p.Run()
+		}()
+	}
+	go func() {
+		wg.Wait()
+		close(successC)
+	}()
+	for isSuccess := range successC {
+		if isSuccess {
```

Comment:
Is the health check already fine if only one of the hosts responds?

Reply:
Yes, as specced

Reply:
I re-read the specs and they say `If frontman doesn't get a positive response from all of them, the internet uplink is considered as unstable and the entire round is skipped` so I think this is not "as specced".